src/D2L.Bmx/UpdateChecker.cs

```diff
@@ -0,0 +1,119 @@
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace D2L.Bmx;
+internal class UpdateChecker {
+
+	public static async Task CheckForUpdatesAsync( BmxConfig config ) {
+		try {
+			string? savedLatestVersion = GetSavedLatestVersion();
+			var localVersion = Assembly.GetExecutingAssembly().GetName().Version;
+			var latestVersion = new Version( savedLatestVersion ?? "0.0.0" );
+			if( ShouldFetchLatestVersion( savedLatestVersion ) ) {
+				latestVersion = new Version( await GetLatestReleaseVersionAsync() );
+			}
+
+			string updateLocation = ( string.Equals( config.Org, "d2l", StringComparison.OrdinalIgnoreCase ) )
+				? "https://bmx.d2l.dev"
+				: "https://github.com/Brightspace/bmx/releases/latest";
+
+			if( latestVersion > localVersion ) {
+				DisplayUpdateMessage( $"A new BMX release is available: v{latestVersion} (current: v{localVersion})\n" +
+					$"Upgrade now at {updateLocation}" );
+			}
+		} catch( Exception ex ) {
+			// Give up and don't bother telling the user we couldn't check for updates
+		}
+	}
+
+	private static void DisplayUpdateMessage( string message ) {
+		var originalBackgroundColor = Console.BackgroundColor;
+		var originalForegroundColor = Console.ForegroundColor;
+
+		Console.BackgroundColor = ConsoleColor.Gray;
+		Console.ForegroundColor = ConsoleColor.Black;
+
+		string[] lines = message.Split( "\n" );
+		int consoleWidth = Console.WindowWidth;
+
+		foreach( string line in lines ) {
+			Console.Error.Write( line.PadRight( consoleWidth ) );
+			Console.Error.WriteLine();
+		}
+
+		Console.BackgroundColor = originalBackgroundColor;
+		Console.ForegroundColor = originalForegroundColor;
+		Console.ResetColor();
+		Console.Error.WriteLine();
+	}
+
+	private static async Task<string> GetLatestReleaseVersionAsync() {
+		using var httpClient = new HttpClient {
+			BaseAddress = new Uri( "https://api.github.com" ),
+			Timeout = TimeSpan.FromSeconds( 5 )
+		};
```

Comment:
```suggestion
		using var httpClient = new HttpClient();
		httpClient.BaseAddress = new Uri( "https://api.github.com" );
		httpClient.Timeout = TimeSpan.FromSeconds( 5 );
```

When you are initializing properties this way it won't properly dispose if there is an error (again unlikely, but lets say you had an exception constructing the Uri object here).

https://stackoverflow.com/a/57204099/4433646

In this particular case (with HttpClient) I don't think it would have mattered anyway (since just constructing an HttpClient instance doesn't grab hold of anything that needs disposal (?)